src/App.tsx

```diff
@@ -15,49 +23,82 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+interface IGoodsQuery {
+  sortBy: string;
+  isReversed: boolean;
+}
+
+const goods: IGoods[] = goodsFromServer.map(name => (
+  {
+    key: getRandomKey(),
+    name,
+  }
+));
+
+function getPreparedGoods(
+  goodsToPrepared: IGoods[],
+  {
+    sortBy,
+    isReversed,
+  }: IGoodsQuery,
+) {
+  const sortedGoods = [...goodsToPrepared];
+
+  if (sortBy) {
+    sortedGoods.sort((goodsA, goodsB) => {
+      switch (sortBy) {
+        case EButtonsSortBy.Alphabet:
+          return goodsA.name.localeCompare(goodsB.name);
+
+        case EButtonsSortBy.Length:
+          return goodsA.name.length - goodsB.name.length;
+
+        default: throw Error('Smth gone wrong');
+      }
+    });
+  }
+
+  if (isReversed) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState(EButtonsSortBy.SortByNone);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const preparedGoods = getPreparedGoods(goods, { sortBy, isReversed });
+
+  const onSortByHandler = (query: string) => {
+    if (EButtonsSortBy.Alphabet === query
+      || EButtonsSortBy.Length === query) {
+      setSortBy(query);
+
+      return;
+    }
+
+    if (EButtonsSortBy.Reset === query) {
+      setSortBy(EButtonsSortBy.SortByNone);
+      setIsReversed(false);
+    }
+  };
```

Comment:
From the name of the function `onSortByHandler`, I can conclude that it is responsible for sorting, but in the middle you also check the condition for resetting everything to default values, which does not exactly correspond to the name.
Therefore, it is worth dividing it into separate functions.